File: projects/project3/common/src/main/java/edu/gvsu/cis/campbjos/imgine/common/model/Results.java
package edu.gvsu.cis.campbjos.imgine.common.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class Results {

    @SerializedName("results")
    @Expose
    private List<Result> resultList;

    public Results() {
        resultList = new ArrayList<>();
    }

    public void addResult(Result result) {
        resultList.add(result);
    }

    public List<Result> list() {
        return resultList;
    }

    public void clear() {
        resultList.clear();
    }

    public void addAll(Results results) {
        if (results == null) {
            return;
        }
        resultList.addAll(results.list());
    }
}

File: projects/project3/common/src/main/java/edu/gvsu/cis/campbjos/imgine/common/model/Results.kt
package edu.gvsu.cis.campbjos.imgine.common.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.util.ArrayList

class Results {

    @SerializedName("results")
    @Expose
    private val resultList: MutableList<Result>

    init {
        resultList = ArrayList<Result>()
    }

    fun addResult(result: Result) {
        resultList.add(result)
    }

    fun list(): List<Result> {
        return resultList
    }

    fun clear() {
        resultList.clear()
    }

    fun addAll(results: Results?) {
        if (results == null) {
            return
        }
        resultList.addAll(results.list())
    }

}
